### ISSUE TITLE
Sendmail mode

### ISSUE THREAD

**Issue by JonZeolla** [rsyslog/rsyslog#1073]
Per http://www.rsyslog.com/doc/v8-stable/configuration/modules/ommail.html, I would like to request sendmail mode as an enhancement to ommail.  Thanks.

**Comment by davidelang** [comment]
"sendmail mode" is a very nebulous thing, which can include a huge number of possible features.

What feature are you needing, and why can't you just run a MTA (sendmail/postfix/exim/etc) on the local box and have it do all the messy stuff?

**Comment by JonZeolla** [comment]
From the documentation:
> The current ommail implementation supports SMTP-direct mode only. In that mode, the plugin talks to the mail server via SMTP protocol. No other process is involved. This mode offers best reliability as it is not depending on any external entity except the mail server.

Thus, my belief was that using a local MTA was not supported.  I am attempting to avoid setting things like `server`, and instead derive the settings from a local MTA.  My apologies for not being specific initially.

**Comment by davidelang** [comment]
there's no real difference between a local mta and a remote one, it's just what address you connect to.

deriving settings from the local mta is extremely unlikely to happen. Do you have any idea how many different MTAs are in common use? or how many different ways there are to configure them to send mail?

the "sendmail mode" would basically be embedding a mta into rsyslog. There would need to be a good justification as to why this was needed ahead of other development work (now, if someone writes something and is talking about contributing it, the bar is lower, but it's still a support question)

**Comment by rgerhards** [comment]
```sendmail``` mode actually means that we would use the external ```sendmail``` program, which would indeed relieve one from knowing local MTA internals.

HOWEVER, given the fact that this is the second requested withion in the 8 years or so, and looking at the priority list, I think we should have many more requests before we begin to work on this. Maybe one could even make ```omprog``` to do that thing (just an idea, I don't have time now to try it out).

**Comment by JonZeolla** [comment]
Thanks, I'll take a look at omprog at some point.  Just wanted to throw my
vote in the pool since it was mentioned on the site and it would be useful
for me to have.  Things are working as is and I plan to migrate away from
ommail to a db insert at some point regardless, just may not be for a few
months.

On Thu, Jun 23, 2016, 01:49 Rainer Gerhards <notifications@github.com>
wrote:

> sendmail mode actually means that we would use the external sendmail
> program, which would indeed relieve one from knowing local MTA internals.
>
> HOWEVER, given the fact that this is the second requested withion in the 8
> years or so, and looking at the priority list, I think we should have many
> more requests before we begin to work on this. Maybe one could even make
> omprog to do that thing (just an idea, I don't have time now to try it
> out).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rsyslog/rsyslog/issues/1073#issuecomment-227956152>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABUkJvaNOw2C718jCSgCN8o2iSmsfuuUks5qOh54gaJpZM4I8Dg8>
> .
>
-- 

Jon

**Comment by rgerhards** [comment]
Yeah, we should probably remove that note now that almost nobody has requested it ;)

**Comment by ymgenesis** [comment]
Is this being developed? I'm trying to get rsyslog to send me email notifications through gmail when a user logs on via sftp. I have this smtp.gmail.com with auth setup in sendmail/postifx (works very well with fail2ban), but can't get ommail to properly do this with the SMTP action settings (failing because it needs authentication I presume). 
Thoughts?

**Comment by davidelang** [comment]
This is not actively being developed. In the 6 years since that comment was 
made, I think you are the second or third person to ask about it.

You shoulad be able to configure your mail server to skip authentication from a 
specific IP address, set it to do that for your rsyslog server and things should 
work.

But it is worth noteing that using rsyslog to generate alerts is extremely 
limiting, you are very quicly going to want more capabilities. So start looking 
at some sort of alerting engine and have rsyslog feed the significant logs to it 
to generate/manage the alerting.

**Comment by febrez** [comment]
+1 for sendmail mode or to specify smtp credentials

**Comment by gnugnug** [comment]
Another +1 for sendmail mode. My SMTP server requires authentication and other things which ommail doesn't support and deploying a full blown MTA as an intermediate is overkill, as every other program I use is able to feed its emails to the sendmail executable.
